test/unit3/test_time.py

```diff
@@ -113,8 +113,11 @@ def test_strftime(self):
         timestamp_to_test = (
             days_to_20020203 * 24 * 60 * 60  # Unix timestamps ignore leap seconds
             + seconds_within_day
-            + time.timezone
         )
+        if time.localtime(timestamp_to_test).tm_isdst:
```

Comment:
In Australia, the 3rd of February is in Daylight Savings Time and hence this test fails by 1 hour without this fix.

Reply:
We should definitely fix this. I just think we should fix the root cause rather than changing the tests.

I have feeling that the problem is that `time.daylight` is returning the incorrect result for the southern hemisphere.

I think if you can fix that then the rest of the tests will pass. Let us know if you need help chasing that bug down.
You might find the pypy/cpython implementation helpful.

https://github.com/mozillazg/pypy/blob/e765b5194c0962ba921a501f59c3011feb7fcf68/pypy/module/time/interp_time.py#L282-L292
https://github.com/python/cpython/blob/main/Modules/timemodule.c#L1781-L1796

Reply:
I think time.daylight is correct for the southern hemisphere. If I `print(time.daylight) `it displays 0 which is correct for today in Australia. The problem is the way the test is written. It assumes that the 3rd of February 2002 is not daylight savings and hence always adds time.timezone (non DST offset). The test needs to add time.timezone in the northern hemisphere and time.altzone in the southern. That's why I updated the test.

Here is how skulpt currently determines if it is daylight savings:

https://github.com/skulpt/skulpt/blob/master/src/lib/time.js#L88-L102

This works because the offset is positive for the southern hemisphere and negative for the northern hemisphere.

Reply:
Really - if I run python on my laptop and switch my timezone to Sydney i get a value of 1 for `time.daylight`

Reply:
Ahh, so it's whether or not we have daylight! I'll fix the time.daylight then. When I print time.daylight in Skulpt I get 0, however on Python I get 1 just like you.

Reply:
All tests now pass for me except this one. :smiley: 

I believe it's because this specific test relies on time.timezone which assumes the chosen date is not currently in daylight savings time, however because we are using localtime(), for me it is.

Reply:
Great - definitely include a fix for this test.

It's not directly from cpython, from what I can gather. And has failed for sometime until a recent pr fixed it (for most people).

Reply:
I've updated the test to use gmtime() instead of localtime() so we don't need to worry about any offsets.